core/drivers/stm32_gpio.c

```diff
@@ -446,6 +446,14 @@ static TEE_Result dt_stm32_gpio_bank(const void *fdt, int node,
 	return TEE_SUCCESS;
 }
 
+static void set_bank_gpio_non_secure(struct stm32_gpio_bank *bank)
+{
+	unsigned int pin = 0;
+
+	for (pin = 0; pin <= bank->ngpios; pin++)
+		stm32_gpio_set_secure_cfg(bank->bank_id, pin, false);
```

Comment:
I was taking a look at I2C driver that uses stm32_gpio_set_secure_cfg() as well. I did not see any deferal sequence while getting the pinctrl, therefore bank/pin.

Am I mistaken?

Reply:
You're right. To move _plat-stm32mp1_ to pintrcl, we need I2C devices to first [get their pinctrl](https://github.com/OP-TEE/optee_os/pull/6050/commits/bb91a0ea9ad4bb6bfaf490eaba3f778bfc349825#diff-fa2bc5bde4570e8b62683401d2d9fbbcd757aa1640238ac80056fc0c7be7334dR743). If that operation fails because it's deferred, current `stm32mp1_pmic.c` i2c bus consumer will report the failure, but that will not happen yet because that driver is initialized at [`driver_init` initcall level](https://github.com/OP-TEE/optee_os/blob/3.21.0/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c#LL643C1-L643C1) while _stm32_gpio.c_ is probed by [_dt_driver.c_ at `early_init_late` initcall level](https://github.com/OP-TEE/optee_os/blob/3.21.0/core/kernel/dt_driver.c#L716).

I agree this is fragile and it is in the plan to move _stm32_i2c.c_ and _stm32mp1_pmic.c_ to dt_driver also for a more consistent init sequence of that platform device drivers. We'll do that in a later P-R.